app/components/trainees/route_indicator/view.rb

```diff
@@ -15,8 +15,34 @@ def render?
         trainee.training_route.present? && trainee.draft?
       end
 
+      def display_text
+        if trainee.apply_application?
+          "Trainee enrolled on #{course_with_code}, #{training_route}"
+        else
+          "Trainee on the #{training_route_link} route".html_safe
+        end
+      end
+
+      def course_with_code
+        "#{trainee.subject}#{course_code}"
+      end
+
+      def course_code
+        return unless course&.code
+
+        " (#{course.code})"
+      end
+
+      def course
+        @course ||= Course.find_by(name: trainee.subject)
```

Comment:
As per @defong 's comment here https://github.com/DFE-Digital/register-trainee-teachers/pull/841#discussion_r629998148 I believe that courses can be duplicated across providers. We potentially need to search by provider too?

Reply:
Updated find to include provider